src/compas_fab/backends/client.py

```diff
@@ -0,0 +1,117 @@
+
+    def get_planning_scene(self, *args, **kwargs):
+        return self.planner.get_planning_scene(*args, **kwargs)
+
+    def add_collision_mesh(self, collision_mesh):
+        return self.planner.add_collision_mesh(collision_mesh)
+
+    def remove_collision_mesh(self, id):
+        return self.planner.remove_collision_mesh(id)
+
+    def append_collision_mesh(self, collision_mesh):
+        return self.planner.append_collision_mesh(collision_mesh)
+
+    def add_attached_collision_mesh(self, attached_collision_mesh):
+        return self.planner.add_attached_collision_mesh(attached_collision_mesh)
+
+    def remove_attached_collision_mesh(self, id):
+        return self.planner.remove_attached_collision_mesh(id)
+
+#     # ==========================================================================
+#     # executing
+#     # ==========================================================================
+#
+#     def execute_joint_trajectory(self, *args, **kwargs):
+#         self.control.execute_joint_trajectory(*args, **kwargs)
+#
+#     def follow_joint_trajectory(self, *args, **kwargs):
+#         self.control.follow_joint_trajectory(*args, **kwargs)
+#
+#     def get_configuration(self, *args, **kwargs):
+#         self.control.get_configuration(*args, **kwargs)
+#
+#
+# class ControlInterface(object):
+#     def execute_joint_trajectory(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+#
+#     def follow_joint_trajectory(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+#
+#     def get_configuration(self, *args, **kwargs):
+#         raise NotImplementedError('Assigned control does not have this feature.')
+
+
+class PlannerInterface(object):
+    def __init__(self, client):
+        self.client = client
+
+    def init_planner(self):
+        pass
+
+    def dispose_planner(self):
+        pass
+
+    def validate_response(self, response):
+        pass
+
+    # ==========================================================================
+    # planning services
+    # ==========================================================================
+
+    def inverse_kinematics(self, *args, **kwargs):
+        raise Exception('Assigned planner does not have this feature.')
+
+    def forward_kinematics(self, *args, **kwargs):
+        raise Exception('Assigned planner does not have this feature.')
+
+    def plan_motion(self, *args, **kwargs):
+        raise Exception('Assigned planner does not have this feature.')
+
+    def plan_cartesian_motion(self, *args, **kwargs):
+        raise Exception('Assigned planner does not have this feature.')
+
+    # ==========================================================================
+    # collision objects and planning scene
+    # ==========================================================================
+
+    def get_planning_scene(self, *args, **kwargs):
```

Comment:
how about moving the planning scene interface methods one level below, so that `PlannerInterface` inherits from it, something like `PlanningSceneInterface` or `CollidableWorldInterface` or similar? This would allow the implementation of tools that only do collision checking but cannot do the higher level planning stuff.

Reply:
I don't think I follow.  The `PlannerInterface` is not creating `abstractmethod`'s in the sense of `ABCMeta` which require implementation in any subclass, but rather is providing default behavior for certain functions which may be overwritten.  So, it would seem that tools that only do collision checking are possible as it is now, and an exception would be raised if the user attempted to call, say, `plan_motion`.

Reply:
Sorry, you are totally right.